File: cpp/dynamic_programming/fibonacci_tabulation.cpp
#include <iostream>

using namespace std;

int fibonacci_Tabulation(int num, int dp[]){
    for(int i=0; i<=num; i++)
    {
        if(i<=1){
            dp[i] = i;
        }
        else
        {
            dp[i] = dp[i-1] + dp[i-2];
        }
    }
    return dp[num];

}

int main(){
    int num, res=0;
    cin>>num;
    int dp[num+1] = {0};
    res = fibonacci_Tabulation(num, dp);
    cout<< res<<endl;
    for(int i=0; i<(sizeof(dp)/sizeof(int)); i++){
        cout << dp[i] <<' ';
    }
}

File: python/dynamic_programming/fibonacci_tabulation.py
def fibonacci_tabulation(num, arr):
	for i in range(0, num+1):
		if i<=1:
			arr.insert(i, i)
		else:
			arr.insert(i, arr[i-1]+arr[i-2])
	return arr[num]



num = int(input())
arr = []
res = fibonacci_tabulation(num, arr)
print(res)
print(arr)
